langroid/agent/chat_agent.py

```diff
@@ -627,6 +644,16 @@ def enable_message(
             if handle:
                 self.llm_tools_handled.add(t)
                 self.llm_functions_handled.add(t)
+
+                # Track whether `output_format` handling was
+                # explicitly enabled and should not be disabled when
+                # `output_format` changes
+                if self.enabled_handling_output_format is not None:
+                    if (
+                        self.enabled_handling_output_format.default_value("request")
+                        == t
+                    ):
+                        self.enabled_handling_output_format = None
```

Comment:
Comment right above the `... = None`:

`t` was designated as "enabled for handling" ONLY for `output_format` enforcement, but we are explicitly enabling it for handling here, so we set the variable to None.